mountpoint-s3/src/data_cache/disk_data_cache.rs

```diff
@@ -176,6 +176,16 @@ impl DiskDataCache {
     }
 }
 
+/// Hash the cache key using its fields, returning the hash encoded as hexadecimal in a new [String].
+fn hash_cache_key(cache_key: &CacheKey) -> String {
```

Comment:
Might make sense to also include the `CACHE_VERSION` here, just as a defense-in-depth thing.